zebrad/tests/acceptance.rs

```diff
@@ -1,30 +1,191 @@
+use std::time::Duration;
+
+/// Executes zebrad binary via `cargo run`.
 pub static RUNNER: Lazy<CmdRunner> = Lazy::new(CmdRunner::default);
 
-/*
- * Disabled pending tracing config rework, so that merging abscissa fixes doesn't block on this
- * test failing because there's tracing output.
- *
-/// Example of a test which matches a regular expression
+#[test]
+fn generate_no_args() {
+    let mut runner = RUNNER.clone();
+    let mut cmd = runner.arg("generate").capture_stdout().run();
+    cmd.stdout()
+        .expect_line("# Default configuration values for zebrad.");
+    cmd.wait().unwrap().expect_success();
+}
+
+#[test]
+fn generate_args() {
+    let mut runner = RUNNER.clone();
+    let cmd = runner
+        .args(&["generate", "argument"])
+        .capture_stdout()
+        .run();
+    cmd.wait().unwrap().expect_code(1);
+
+    // Invalid flag
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["generate", "-f"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    // Valid flag but missing argument
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["generate", "-o"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    let mut runner = RUNNER.clone();
+    let cmd = runner
+        .args(&["generate", "-o", "file"])
+        .capture_stdout()
+        .run();
+    cmd.wait().unwrap().expect_success();
+
+    // Todo: Check if the file was created
+}
+
+#[test]
+fn help_no_args() {
+    let mut runner = RUNNER.clone();
+    let cmd = runner.arg("help").capture_stdout().run();
+    cmd.wait().unwrap().expect_success();
+}
+
+#[test]
+fn help_args() {
+    // Invalid argument
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["help", "argument"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    // Invalid flag
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["help", "-f"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+}
+
+#[test]
+fn revhex_no_args() {
+    let mut runner = RUNNER.clone();
+    let _cmd = runner
+        .timeout(Duration::from_secs(1))
+        .arg("revhex")
+        .capture_stdout()
+        .run();
+
+    // Program is waiting for input, we just exit after 1 second
+}
+
+#[test]
+fn revhex_args() {
+    let mut runner = RUNNER.clone();
+    let mut cmd = runner.args(&["revhex", "33eeff55"]).capture_stdout().run();
+    cmd.stdout().expect_line("55ffee33");
+}
+
+#[test]
+fn seed_no_args() {
+    let mut runner = RUNNER.clone();
+    runner.exclusive();
+    let mut cmd = runner
+        .timeout(Duration::from_secs(1))
+        .arg("seed")
+        .capture_stdout()
+        .run();
+
+    // Todo: maybe add special info!() to seed command
+    // Todo: improve the regex
+    cmd.stdout()
+        .expect_regex(r"^(.*?)Initializing tracing endpoint");
+
+    // Problem: Zombie child process is not killed
+    //cmd.kill();
```

Comment:
On Unix-based operating systems (Linux, macOS), zombie processes stick around until the parent process calls `wait`:
https://docs.rs/abscissa_core/0.5.2/abscissa_core/testing/process/struct.Process.html#method.wait

Reply:
The `wait()` haves an error as we made the process time out. I added `cmd.wait().unwrap_err();` at https://github.com/ZcashFoundation/zebra/pull/690/commits/1d05af32b82f6ee49ffa50ef827953ba7b99364f however the process is still around. Maybe i am not calling it correctly ?